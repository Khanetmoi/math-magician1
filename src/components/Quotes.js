import React, { useState, useEffect } from 'react';
import { v4 as id } from 'uuid';


const loadingMessage = <p>Waiting for data loading....</p>;
const errorMessage = <p>Oops, something went wrong!!!</p>;

const Quotes = () => {
  const [quotes, setQuotes] = useState(null);
  const [isLoading, setIsLoading] = useState(false);
  const [hasError, setHasError] = useState(false);

  const quotesApi = async () => {
    setIsLoading(true);
    try {
      fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
        headers: {
          'X-Api-Key': 'FSvXTjxiPe8a+r73Jt+dLQ==j1LfG5Zfc7QrFBl8',
        },
      })
        .then((Response) => Response.json())
        .then((data) => {
          setQuotes(data);
        });
    } catch (error) {
      setHasError(true);
    }
    setIsLoading(false);
  };

  useEffect(() => {
    quotesApi();
  }, [setQuotes, setIsLoading]);
  return (
    <div className="quotesbox">
      {isLoading ? loadingMessage : null}
      {hasError ? errorMessage : null}
      {quotes && Array.isArray(quotes)
        && quotes.map((data) => (
          <div key={id()}>
            <h2>
              {data.quote}
              {' '}
              -
              {' '}
              {data.author}
            </h2>
          </div>
        ))}
    </div>
  );
};


export default Quotes;