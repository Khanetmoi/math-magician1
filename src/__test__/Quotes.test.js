import React from "react";
import renderer from 'react-test-renderer';
import Quotes from "../components/Quotes";
import '@testing-library/jest-dom';

describe('Test of Quote page', () => {
    it('test if Quote page is rendered', () => {
        const quoteTest = renderer
            .create(<Quotes />)
            .toJSON();
        expect(quoteTest).toMatchSnapshot();
    });
});