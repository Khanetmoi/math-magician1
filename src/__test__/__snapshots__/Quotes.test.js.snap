// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Quote page test if Quote page is rendered 1`] = `
<div
  className="quotesbox"
/>
`;
